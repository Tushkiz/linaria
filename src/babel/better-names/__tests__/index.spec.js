/* eslint-disable no-template-curly-in-string */
/* @flow */

import * as babel from 'babel-core';
import path from 'path';
import dedent from 'dedent';

function transpile(source, options) {
  return babel.transform(source,
    {
      plugins: [
        require.resolve('../index')
      ],
      babelrc: false,
      ...options
    },
  ).code;
}

it('should convert `css` to `css.named` with filename', () => {
  const env = process.env.NODE_ENV
  process.env.NODE_ENV = 'development';
  const code = transpile(dedent`
  const header = css\`
    font-size: 3em;
  \`;
  `, { filename: 'Header.js' });

  expect(code).toMatchSnapshot();
  process.env.NODE_ENV = env;
});

it('should convert `css` to `css.named` without filename', () => {
  const env = process.env.NODE_ENV
  process.env.NODE_ENV = 'production';
  const code = transpile(dedent`
  const header = css\`
    font-size: 3em;
  \`;
  `, { filename: 'Header.js' });

  expect(code).toMatchSnapshot();
  process.env.NODE_ENV = env;
});


it('should convert `css` to `css.named` for multiple literals', () => {
  const code = transpile(dedent`
  const header = css\`
    font-size: ${'${2 + 1}'}em;
  \`;

  const body = css\`
    border-radius: ${'${2 + 2}'}px;
  \`;
  `);

  expect(code).toMatchSnapshot();
});

it('should not convert `css.named()` literal', () => {
  const code = transpile(dedent`
  const header = css.named('my-header')\`
    font-size: 3em;
  \`;
  `);

  expect(code).toMatchSnapshot();
});

it('should convert non-top-level `css` to `css.named`', () => {
  const code = transpile(dedent`
  const defaults = {
    fontSize: '3em',
  };

  function render() {
    const header = css\`
      font-size: ${'${defaults.fontSize}'};
    \`;
  }
  `);

  expect(code).toMatchSnapshot();
});

it('should not convert random tags', () => {
  const code = transpile(dedent`
  const header = scss\`
    font-size: 3em;
  \`;
  `, { filename: 'Header.js' });

  expect(code).toMatchSnapshot();
});
