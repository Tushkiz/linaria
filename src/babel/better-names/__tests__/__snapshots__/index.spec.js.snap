// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert \`css\` to \`css.named\` for multiple literals 1`] = `
"const header = css.named(\\"header\\", \\"unknown\\")\`
  font-size: \${2 + 1}em;
\`;

const body = css.named(\\"body\\", \\"unknown\\")\`
  border-radius: \${2 + 2}px;
\`;"
`;

exports[`should convert \`css\` to \`css.named\` with filename 1`] = `
"const header = css.named(\\"header\\", \\"Header.js\\")\`
  font-size: 3em;
\`;"
`;

exports[`should convert \`css\` to \`css.named\` without filename 1`] = `
"const header = css.named(\\"header\\")\`
  font-size: 3em;
\`;"
`;

exports[`should convert non-top-level \`css\` to \`css.named\` 1`] = `
"const defaults = {
  fontSize: '3em'
};

function render() {
  const header = css.named('header', 'unknown')\`
    font-size: \${defaults.fontSize};
  \`;
}"
`;

exports[`should not convert \`css.named()\` literal 1`] = `
"const header = css.named('my-header')\`
  font-size: 3em;
\`;"
`;

exports[`should not convert random tags 1`] = `
"const header = scss\`
  font-size: 3em;
\`;"
`;
